### PR TITLE
Enable nullability in DomainUpDown accessible objects

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in DomainUpDown accessible objects.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6357)

### CODE REVIEW

**Comment by dreddy-work** [comment]
@gpetrou , unit tests failed, please take a look. @vladimir-krestov  as FYI.

**Comment by dreddy-work** [comment]
@SergeySmirnov-Akvelon , can you or your team check on accessible test changes here?

File: src/System.Windows.Forms/src/System/Windows/Forms/DomainUpDown.DomainItemAccessibleObject.cs [hunk 1]
```diff
@@ -2,40 +2,38 @@
...
+            private string? _name;
+            private readonly DomainItemListAccessibleObject _parent;
 
-            public DomainItemAccessibleObject(string name, AccessibleObject parent) : base()
+            public DomainItemAccessibleObject(string? name, AccessibleObject parent) : base()
             {
-                this.name = name;
-                this.parent = (DomainItemListAccessibleObject)parent;
+                _name = name;
+                _parent = (DomainItemListAccessibleObject)parent.OrThrowIfNull();
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/DomainUpDown.DomainItemAccessibleObject.cs:17]: @dreddy-work this is a breaking change that needs to documented once thus is merged.

**Review by vladimir-krestov — APPROVED** [review]
Approved ✔️

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**